src/charm.py

```diff
@@ -343,6 +346,41 @@ def _pebble_checks(self, otelcol_args: List[str]) -> Dict[str, CheckDict]:
         }
         return checks
 
+    def _write_ca_certificates_to_disk(self, scrape_jobs: List[Dict], container: Container) -> Dict[str, str]:
+        cert_paths = {}
+        jobs_with_certs = []
+
+        for job in scrape_jobs:
+            tls_config = job.get("tls_config", {})
+            ca_file_content = tls_config.get("ca_file")
+
+            if ca_file_content and ca_file_content.strip().startswith("-----BEGIN CERTIFICATE-----"):
+                jobs_with_certs.append(job)
+
+        if not jobs_with_certs:
+            return cert_paths
+
+        if not container.can_connect():
+            logger.warning("Container not accessible, skipping CA certificate processing")
+            return cert_paths
+
+        # Ensure certs dir.
+        container.exec(["mkdir", "-p", "/etc/ssl/certs/"]).wait()
+
+        for job in jobs_with_certs:
+            tls_config = job.get("tls_config", {})
+            ca_file_content = tls_config.get("ca_file")
```

Comment:
We already have this information within the previous for-loop. Should we build cert_paths at the same time we're checking for `.startswith(...)`?